Add paperless_ngx docs (#39103)
home-assistant/home-assistant.github.io: source/_integrations/paperless_ngx.markdown

---
title: Paperless-ngx
description: Instructions on how to integrate Paperless-ngx into Home Assistant

> ha_release: 2025.7

Reviewer: ```suggestion
ha_release: 2025.6
```

ha_category:
  - Sensor
ha_iot_class: Local Polling
ha_config_flow: true
ha_domain: paperless_ngx
ha_codeowners:
  - '@fvgarrel'
ha_integration_type: service
ha_quality_scale: bronze
related:
  - url: https://docs.paperless-ngx.com/
    title: Paperless-ngx
---

The **Paperless-ngx** {% term integration %} allows you to connect your [Paperless-ngx](https://docs.paperless-ngx.com/) instance to Home Assistant and monitor its status and activity.

## Prerequisites

To ensure full functionality of this integration, you must have **read permissions** for all document-related resources, including documents, tags, document types, and correspondents.

> To enable monitoring of diagnostic sensors, you must have **administrator permissions**. Without administrator rights, specific API endpoints cannot be accessed, and the sensor states will not be available.

Reviewer: this is not implemented right?

Author: Yes, it was in the first version of my PR. I will remove it.

---
title: Paperless-ngx
description: Instructions on how to setup the Paperless-ngx integration
ha_release: 2025.7
ha_category:
  - Sensor
ha_iot_class: Local Pull
ha_config_flow: true
ha_domain: paperless_ngx
ha_codeowners:
  - '@fvgarrel'
ha_integration_type: service
ha_quality_scale: bronze
related:
  - url: https://docs.paperless-ngx.com/
    title: Paperless-ngx
---

The **Paperless-ngx** {% term integration %} allows you to connect your [Paperless-ngx](https://docs.paperless-ngx.com/) instance to Home Assistant and monitor its status and activity.

## Prerequisites

> To ensure full functionality of this integration, the user must have **read permissions** for all document-related resources, including documents, tags, document types, and correspondents.

Reviewer 2: ```suggestion
To ensure full functionality of this integration, you must have **read permissions** for all document-related resources, including documents, tags, document types, and correspondents.
```
we prefer the direct address "you" over "the user"

> To enable monitoring of diagnostic sensors, the user must have **admin permissions**. Without admin rights, specific API endpoints cannot be accessed, and the sensor states will not be available.

Reviewer 2: ```suggestion
To enable monitoring of diagnostic sensors, you must have **administrator permissions**. Without administrator rights, specific API endpoints cannot be accessed, and the sensor states will not be available.
```

{% details "Create an access token" %}

1. Log in to your **Paperless-ngx** instance.

> 2. Click on your profile icon in the upper-right corner.

Reviewer 2: ```suggestion
2. In the upper-right corner, select your profile icon.
```

3. Select **My Profile**.

> 4. Under **API Auth Token**, click the right **Refresh** button next to the textbox to generate a new token. Confirm with **yes**.

Reviewer 2: ```suggestion
4. Under **API Auth Token**, select the right **Refresh** button next to the textbox to generate a new token. Confirm with **yes**.
```

5. Copy the token and use it during the integration setup in Home Assistant.

{% enddetails %}

{% include integrations/config_flow.md %}

{% configuration_basic %}
Host:
  description: "The host to connect to"
Access token:
  description: "The access token for Paperless-ngx user."
{% endconfiguration_basic %}

## Sensors

This integration provides {% term sensors %} for the following information from Paperless-ngx:

- Number of documents
- Number of document with inbox tags
- Status of the database with errors
- Status of Redis with errors
- Status of Celery
- Status of classifier with errors
- Status of index with errors

> - Storage Total
> - Storage Available

Reviewer 2: ```suggestion
- Storage total
- Storage available
```
could these use sentence-style capitalization, like other UI strings?
